src/smccc.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Decoder for SMC Calling Convention ARM DEN 0028E v1.4
+
+mod smccc_arm;
+mod smccc_common;
+mod smccc_hyp;
+mod smccc_secure;
+mod smccc_tapp;
```

Comment:
Having an `smccc_` prefix for every submodule under the `smccc` module seems unnecessarily repetitive, how about just calling them `arm`, `common` and so on?